src/sentry/api/endpoints/organization_projects_experiment.py

```diff
@@ -0,0 +1,188 @@
+import logging
+import random
+import string
+
+from django.db import IntegrityError, transaction
+from rest_framework.exceptions import PermissionDenied
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.serializers import ValidationError
+
+from sentry import audit_log, features
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.organization import OrganizationEndpoint, OrganizationPermission
+from sentry.api.endpoints.team_projects import ProjectSerializer
+from sentry.api.exceptions import ConflictError, ResourceDoesNotExist
+from sentry.api.serializers import serialize
+from sentry.experiments import manager as expt_manager
+from sentry.models import Project
+from sentry.models.organization import Organization
+from sentry.models.organizationmember import OrganizationMember
+from sentry.models.organizationmemberteam import OrganizationMemberTeam
+from sentry.models.team import Team
+from sentry.signals import project_created, team_created
+from sentry.utils.snowflake import MaxSnowflakeRetryError
+
+CONFLICTING_TEAM_SLUG_ERROR = "A team with this slug already exists."
+MISSING_PERMISSION_ERROR_STRING = "You do not have permission to join a new team as a Team Admin."
+
+
+def generate_three_letter_string():
+    letters = string.ascii_lowercase
+    return "".join(random.choice(letters) for _ in range(3))
+
+
+# This endpoint is intended to be available to all members of an
+# organization so we include "project:read" in the POST scopes.
+
+
+class OrgProjectPermission(OrganizationPermission):
+    scope_map = {
+        "POST": ["project:read", "project:write", "project:admin"],
+    }
+
+
+@region_silo_endpoint
+class OrganizationProjectsExperimentEndpoint(OrganizationEndpoint):
+    permission_classes = (OrgProjectPermission,)
+    logger = logging.getLogger("team-project.create")
+
+    def should_add_creator_to_team(self, request: Request):
+        return request.user.is_authenticated
+
+    def post(self, request: Request, organization: Organization) -> Response:
+        """
+        Create a new Team and Project
```

Comment:
What's the use case in creating both a team _and_ a project?

Is there any reason why we've bundled this together in a single endpoint, rather than 2 separate calls?

I'm imagining we would also want the ability to add a project to a team _without_ creating a new team, as well as the ability to create a new team _without_ necessarily creating a new project?

Reply:
@nhsiehgit this is not a permanent endpoint and not the only endpoint for project creation. We're still going to use the old one for most cases, but only when someone has no team-admin/org permissions they create a project through this endpoint. We create their own team and from that point they can create projects for that team (through existing non-experimental endpoint). So basically this is just a hacky workaround.